src/repositoryModals.jsx

```diff
@@ -0,0 +1,451 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect, useState } from 'react';
+import PropTypes from "prop-types";
+
+import cockpit from 'cockpit';
+
+import { Alert } from "@patternfly/react-core/dist/esm/components/Alert";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Checkbox } from '@patternfly/react-core/dist/esm/components/Checkbox';
+import { Flex, FlexItem } from "@patternfly/react-core/dist/esm/layouts/Flex/index.js";
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { Modal } from "@patternfly/react-core/dist/esm/components/Modal";
+import { Text } from "@patternfly/react-core/dist/esm/components/Text";
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { TextArea } from "@patternfly/react-core/dist/esm/components/TextArea";
+import { Select, SelectOption } from "@patternfly/react-core/dist/esm/deprecated/components/Select";
+import { Spinner } from "@patternfly/react-core/dist/esm/components/Spinner/index.js";
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+
+import { FormHelper } from 'cockpit-components-form-helper.jsx';
+import { useDialogs } from "dialogs.jsx";
+
+import * as remotes from './remotes';
+
+const _ = cockpit.gettext;
+
+export const RemoveRepositoryModal = ({ origin, availableRemotes, refreshRemotes }) => {
+    const Dialogs = useDialogs();
+    const [error, setError] = useState('');
+    const [selectedRemotes, setSelectedRemotes] = useState([]);
+
+    const onDelete = () => {
+        Promise.all(selectedRemotes.map(remote => remotes.deleteRemote(remote)))
+                .then(() => refreshRemotes())
+                .then(Dialogs.close, ex => setError(ex.message));
+    };
+
+    const handleChange = (event, isChecked) => {
+        if (isChecked) {
+            setSelectedRemotes([...selectedRemotes, event.target.id]);
+        } else {
+            setSelectedRemotes(selectedRemotes.filter(remote => remote !== event.target.id));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231124-162305-47d51597-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)